HashSet.java

```diff
@@ -0,0 +1,51 @@
+// Time Complexity : O(N) for searching and remove. O(1) for insertion as I am adding new elemnet at head.
+// Space Complexity : O(N)
+// Did this code successfully run on Leetcode :
```

Comment:
Please run this code on leetcode and fill this space

Reply:
I did updated the code and pushed my comment on github. Should I raise another pull request ?

Reply:
No, you don't have to create a new PR.